File: Sparkmax/SparkmaxJoystickButtonJava/src/main/java/frc/robot/Robot.java
package frc.robot;

// import Joystick (VS code will automatically do this after you declare your joystick)
import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.TimedRobot;
//0. import PWMSparkMax (VS code will automatically do this after you declare your motor)
import edu.wpi.first.wpilibj.motorcontrol.PWMSparkMax;

public class Robot extends TimedRobot {
  // declare the joystick object
  public Joystick joystick;
  // declare PWMSpark max motor
  public PWMSparkMax basicMotor;
  

  @Override
  public void robotInit() {
    //Initialize the joystick to port 0
    joystick = new Joystick(0);
    //Initialize basic motor to use PWM channel Zero
    basicMotor = new PWMSparkMax(0);
  }

  @Override
  public void teleopPeriodic() {
    //if joystick button 1 is pressed
    if (joystick.getRawButton(1) == true) {
      // Run the motor at 100% speed clockwise
      basicMotor.set(-1.0);
    } 
    else {
      //if button isn't pressed, stop the motor
      basicMotor.set(0.0);
    }
  }
}

File: Sparkmax/SparkmaxJoystickButtonCpp/src/main/cpp/Robot.cpp
#include "Robot.h"

#include <fmt/core.h>
//1. include Joystick library
#include "frc/Joystick.h"
//2. include PWMSparkMax library
#include "frc/motorcontrol/PWMSparkMax.h"

//3. declare joystick object
frc::Joystick joystick{0};
//4. declare basic motor object with PWM port number inbetween {}
frc::PWMSparkMax basicMotor{0};

void Robot::TeleopPeriodic() {
  if (joystick.GetRawButton(1) == true) {
    //if button 1 is pressed, turn on motor full speed clock wise
    basicMotor.Set(-1.0);
  } else {
    //else stop motor
    basicMotor.Set(0);
  }
}

void Robot::RobotInit() {
}

void Robot::RobotPeriodic() {}

void Robot::AutonomousInit() {
}

void Robot::AutonomousPeriodic() {
}

void Robot::TeleopInit() {}

void Robot::DisabledInit() {}

void Robot::DisabledPeriodic() {}

void Robot::TestInit() {}

void Robot::TestPeriodic() {}

void Robot::SimulationInit() {}

void Robot::SimulationPeriodic() {}

#ifndef RUNNING_FRC_TESTS
int main() {
  return frc::StartRobot<Robot>();
}
#endif
